refactor: update `math/base/special/hypot` to follow latest project conventions

vivekmaurya001 opened the pull request:
Resolves None.

## Description

- refactored main.c to use stdlib api instead of math.h for  `math/base/special/hypot` 
- updated benchmark calculations
- updated js examples
- updated test file
- updated readme

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
refactor: update math/base/special/hypot to follow latest project conventions

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 commented:
@Planeshifter , @gunjjoshi , please give this a review

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/hypot/benchmark/benchmark.js:
```diff
@@ -21,8 +21,8 @@
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
```
@vivekmaurya001 In the current JS benchmarks (https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/hypot/benchmark/benchmark.js#L47), we are not using integers. We can use `@stdlib/random/array/uniform` here. Or is there any specific reason for using `discrete-uniform`?

vivekmaurya001 replied:
actually in examples in index.js it is getting round thats why  i thought it , i will correct it soon

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/hypot/benchmark/benchmark.native.js:
```diff
@@ -22,8 +22,8 @@
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var discreteUniform = require( '@stdlib/random/array/discrete-uniform' );
```
Same comment here.

vivekmaurya001 replied:
i will correct it soon

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/hypot/benchmark/c/native/benchmark.c:
```diff
@@ -91,17 +91,20 @@ static double rand_double( void ) {
 	double t;
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = round( ( rand_double() * 100.0 ) - 50.0 );
+		y[ i ] = round( ( rand_double() * 100.0 ) - 50.0 );
+	}
```
Same comment here, regarding the usage of `round`.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/hypot/src/hypot.c:
@vivekmaurya001 Can you please change the name of this file to `main.c`?

gunjjoshi replied:
Along with that, you'll also have to modify the `manifest.json` file to use "./src/main.c" instead of using "./src/hypot.c".

vivekmaurya001 replied:
ook

vivekmaurya001 committed:
refactor: update math/base/special/hypot to follow latest project conventions

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 commented:
done the changes @gunjjoshi , i will do the same changes in my other related PR's

kgryte committed:
docs: update example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM. Thanks for reviewing, @gunjjoshi!
